tests/test_leftovers.py

```diff
@@ -0,0 +1,294 @@
+from indexd.blueprint import get_record
+from tests.test_client import get_doc
+import pytest
+from indexd.index.errors import NoRecordFound as IndexNoRecordFound
+from indexd.dos.blueprint import get_dos_record
+import json
+
+
+def test_get_record_call_dist_get_record(client, user):
+    # test get_record correctly
+    data = get_doc()
+    res = client.post("/index/", json=data, headers=user)
+    assert res.status_code == 200
+    rec = res.json
+    response = get_record(rec["did"])
+    record = response[0].json
+    assert record["urls"][0] == "s3://endpointurl/bucket/key"
+    assert record["size"] == 123
+
+    # test call dist_get_record for unposted record
+    fake_did = "testprefix:d96bab16-c4e1-44ac-923a-04328b6fe78f"
+    with pytest.raises(IndexNoRecordFound):
+        get_record(fake_did)
+
+
+def test_get_dos_record(client, user):
+    data = get_doc(has_urls_metadata=True, has_metadata=True, has_baseid=True)
+
+    res_1 = client.post("/index/", json=data, headers=user)
+    assert res_1.status_code == 200
+
+    # test exception raised at nonexistent
+    fake_did = "16e51a9a-af23-45a3-b4b2-94ae2b143feb"
+    with pytest.raises(IndexNoRecordFound):
+        get_dos_record(fake_did)
+
+
+def test_alias_list_by_hash(client, user):
+    data = {
+        "size": 123,
+        "hashes": {"md5": "8b9942cf415384b27cadf1f4d2d682e5"},
+        "release": "private",
+        "keeper_authority": "CRI",
+        "host_authorities": ["PDC"],
+    }
+    ark = "ark:/31807/TEST-abc"
+    res = client.put("/alias/" + ark, json=data, headers=user)
+    assert res.status_code == 200
+    rec = res.json
+    assert rec["name"] == ark
+    # assert there is only one entry in the alias index
+    assert len(client.get("/alias/").json["aliases"]) == 1
+    res = client.get("/alias/?hash=" + "md5:8b9942cf415384b27cadf1f4d2d682e5")
+    rec = res.json
+    # assert that the returned alias by hash is the same as the posted
+    assert rec["aliases"][0] == ark
+    assert rec["hashes"] == data["hashes"]
+
+
+def test_alias_list_by_size(client, user):
+    data = {
+        "size": 123,
+        "hashes": {"md5": "8b9942cf415384b27cadf1f4d2d682e5"},
+        "release": "private",
+        "keeper_authority": "CRI",
+        "host_authorities": ["PDC"],
+    }
+    ark = "ark:/31807/TEST-abc"
+    res = client.put("/alias/" + ark, json=data, headers=user)
+    assert res.status_code == 200
+    rec = res.json
+    assert rec["name"] == ark
+    # assert there is only one entry in the alias index
+    assert len(client.get("/alias/").json["aliases"]) == 1
+    # assert that the returned alias by size is the same as the posted
+    res = client.get("/alias/?size=" + str(data["size"]))
```

Comment:
using `format` to build strings lets you use ints without having to convert to strings :D